### ISSUE TITLE
how to set up default parameters

### ISSUE THREAD

**Issue by BenKaehler** [qiime2/q2cli#44]
@ebolyen I'm using [q2-feature-classifier](https://github.com/BenKaehler/q2-feature-classifier) as an example for this issue. I'm not sure if I've set it up properly to give a method parameter a default value, so please let me know if I haven't.

Method signature:

```
def classify(sequences : types.GeneratorType, reference_taxonomy : pd.Series,
        reference_sequences : types.GeneratorType, method : str,
        depth : int = 4) -> pd.Series:
```

Method registration:

```
plugin.methods.register_function(
    function=q2_feature_classifier.classify,
    inputs={'sequences' : FeatureData[Sequence],
        'reference_sequences' : ReferenceFeatures[SSU],
        'reference_taxonomy' : ReferenceFeatures[SSU]},
    parameters={'depth': Int,
                'method' : Str % Choices(['naive-bayes', 'svc', 'perfect'])},
    outputs=[('classification', FeatureData[Taxonomy])],
    name='Train and apply feature classifier.',
    description='Train a classifier and apply it to feature data.'
)
```

Unexpected behaviour:

```
$ wget https://github.com/qiime2/q2-types/raw/master/q2_types/tests/data/reference-features-ssu.qza
$ wget https://github.com/qiime2/q2-types/raw/master/q2_types/tests/data/feature-data-sequence.qza
$ qiime feature-classifier classify --i-reference-sequences reference-features-ssu.qza --i-reference-taxonomy reference-features-ssu.qza --i-sequences feature-data-sequence.qza --p-method perfect --o-classification classification.qza --p-depth 5
$ qiime feature-classifier classify --i-reference-sequences reference-features-ssu.qza --i-reference-taxonomy reference-features-ssu.qza --i-sequences feature-data-sequence.qza --p-method perfect --o-classification classification.qza
Usage: qiime feature-classifier classify [OPTIONS]

Error: Missing option "--p-depth".
```

So it fails when I don't give it the `depth` parameter.

Thanks!

**Comment by ebolyen** [comment]
Thanks @BenKaehler I think this code *should* work. We will probably need to modify the way interfaces interact with the framework to support this. `Signature` is an obvious target, perhaps it should have some way of stating parameters are optional?

**Comment by BenKaehler** [comment]
Thanks @ebolyen, can I do anything for that, or should I leave it with you?

**Comment by ebolyen** [comment]
It is probably easier if we do this, but I can't guarantee any timeline for it. That being said, if you wanted to help out I am happy to point you to what would probably need to change, just message me on Slack whenever and we can talk in more detail, our codebase is still pretty rough around the edges.

**Comment by jairideout** [comment]
Fixed in #54.
